packages/vm/src/state/stateManager.ts

```diff
@@ -603,7 +603,14 @@ export default class DefaultStateManager implements StateManager {
   async accountExists(address: Address): Promise<boolean> {
     const account = this._cache.lookup(address)
     if (account && !this._cache.keyIsDeleted(address)) {
-      return true
+      if (account.balance.isZero() && !this._common.gteHardfork('spuriousDragon')) {
```

Comment:
A more complete check here is to use `account.isEmpty()` (this would return `true` for contracts which have 0 balance :) )

Reply:
Thanks!  Updated in the latest commit.